### PR TITLE
Printing tickets on character printers connected to network.

### PR DESCRIPTION
**Pull request by mezkov**
I've added possibility to print tickets on Epson character printers that are connected to network instead of LPT. This was pretty easy, because data format accepted by Epson printers on TCP port 9100 is the same as on LPT. 

I've also added Epson TM-U220B to the models combo box on printer options dialog and made it change the line length to 40 characters when selected.

### CODE REVIEW

File: quickevent/app/plugins/qml/Receipts/src/receiptsprinteroptions.h [hunk 1]
```diff
@@ -17,8 +17,15 @@ class ReceiptsPrinterOptions : public QVariantMap
 	QF_VARIANTMAP_FIELD2(QString, c, setC, haracterPrinterModel, QStringLiteral("Epson TM-T88V"))
 	QF_VARIANTMAP_FIELD2(int, c, setC, haracterPrinterLineLength, 41)
 	QF_VARIANTMAP_FIELD2(bool, is, set, CharacterPrinterGenerateControlCodes, true)
+	QF_VARIANTMAP_FIELD(int, c, setC, haracterPrinterType)
+	QF_VARIANTMAP_FIELD(QString, c, setC, haracterPrinterAddress)
 public:
 	enum class PrinterType : int {GraphicPrinter = 0, CharacterPrinter};
+	enum CharacterPrinteType {
```
Comment (fvacek) [inline quickevent/app/plugins/qml/Receipts/src/receiptsprinteroptions.h]: please use enum class without explicit values set in this case
```c++
enum CharacterPrinteType {
	LPT,
	Directory,
	Network,
	};
```
```

File: quickevent/app/plugins/qml/Receipts/src/receiptsprinter.cpp [hunk 2]
```diff
@@ -110,17 +111,51 @@ void ReceiptsPrinter::printReceipt(const QString &report_file_name, const QVaria
...
+			}
+			case ReceiptsPrinterOptions::CharacterPrinteType::LPT: {
+				if (!printer_opts.characterPrinterDevice().isEmpty()) {
+					save_file(printer_opts.characterPrinterDevice());
+				}
+				break;
+			}
+			case ReceiptsPrinterOptions::CharacterPrinteType::Network: {
+				if (!printer_opts.characterPrinterAddress().isEmpty()) {
+					QTcpSocket socket;
```
Comment (fvacek) [inline quickevent/app/plugins/qml/Receipts/src/receiptsprinter.cpp]:
```c++
QString host = printer_opts.characterPrinterAddress().section(':', 0, 0);
int port = printer_opts.characterPrinterAddress().section(':', 1, 1).toInt();
if(port == 0)
    port = 9100;
```

**Comment by fvacek** [comment]
thanks, review is started

**Comment by mezkov** [comment]
done

**Pull request merged**